servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/MulticastPublisher.java

```diff
@@ -179,32 +239,26 @@ void addSubscriber(Subscriber<? super T> subscriber, ContextMap contextMap,
             }
             MulticastFixedSubscriber<T> multiSubscriber =
                     new MulticastFixedSubscriber<>(this, subscriber, contextMap, contextProvider, sCount);
-            for (;;) {
-                final Subscriber<? super T>[] currSubs = subscribers;
-                if (currSubs.length == 1 && currSubs[0] instanceof TerminalSubscriber) {
-                    ((TerminalSubscriber<?>) currSubs[0]).terminate(subscriber);
-                    break;
-                } else {
-                    @SuppressWarnings("unchecked")
-                    Subscriber<? super T>[] newSubs = (Subscriber<? super T>[])
-                            Array.newInstance(Subscriber.class, currSubs.length + 1);
-                    System.arraycopy(currSubs, 0, newSubs, 0, currSubs.length);
-                    newSubs[currSubs.length] = multiSubscriber;
-                    if (newSubscribersUpdater.compareAndSet(this, currSubs, newSubs)) {
-                        addSubscriber(multiSubscriber);
```

Comment:
note this code was moved to `addSubscriber` bcz replay requires that we acquire the lock before making the subscriber visible in the array (via `newSubscribersUpdater`) and this is preferred approach for multicast too.